python/servo/mutation/README.md

```diff
@@ -0,0 +1,55 @@
+﻿## Implement Mutation Testing on Servo Parallel Browsing Project
+
+
+The motivation for mutation testing is to test the breadth coverage of tests for source code. Faults (or mutations) are automatically seeded into the code, then tests are run. If tests fail then the mutation is killed, if the tests pass then the mutation lived. The quality of tests can be gauged from the percentage of mutations killed.
+
+For more info refer [Wiki page](https://en.wikipedia.org/wiki/Mutation_testing).
+
+In this project, mutation testing is used to test the coverage of WPT for Servo's browsing engine.
+
+### Mutation Strategy
+This version of mutation testing consists of a Python script that finds random uses of && in Servo's code base and replaces them by ||. The expectation from the WPT tests is to catch this mutation and result in failures when executed on the corresponding code base.
+
+### Test Run Strategy
+The mutation test aims to run only tests which are concerned with the mutant. Therefore part of WPT test is related to the source code under mutation is invoked. For this it requires a test mapping in source folders.
+
+#### test_mapping.json
+The file test_mapping.json is used to map the source code to their corresponding WPT tests. The user must maintain a updated version of this file in the path where mutation testing needs to be performed. Additionally, the test_mapping.json will only consist of maps of source codes that are present in the current directory. Hence, each folder will have a unique test_mapping.json file. Any source code files that may be present in a path but are not mapped to a WPT in test_mapping.json will not be covered for mutation testing.
+
+### Sample test_mapping.json format
+A sample of test_mapping.json is as shown below:
+
+```
+{
+  "xmlhttprequest.rs": [
+    "XMLHttpRequest"
+  ],  
+  "range.rs": [
+    "dom/ranges"
+  ]
+}
+```
+
+Please ensure that each folder that requires a mutant to be generated consists of test_mapping.json file so that the script can function as expected.

```

Comment:
Are wildcards allowed? If so, can we have an example of one?

Reply:
Wildcards are not allowed.

Reply:
Please mention this in the readme; nobody wondering later will be able to easily find this comment :)